digdag-ui/model.js

```diff
@@ -252,12 +252,11 @@ export class Model {
     return this.get(`logs/${attemptId}/files?task=${encodeURIComponent(taskName)}`)
   }
 
-  fetchLogFile (file: LogFileHandle) {
+  fetchLogFile (attemptId: number, file: LogFileHandle) {
     if (!file.direct) {
-      return new Promise((resolve, reject) => {
-        reject(`Cannot fetch non-direct log file: ${file.fileName}`)
-      })
+      file.direct = this.config.url + `logs/${attemptId}/files/${file.fileName}`
```

Comment:
I don't like the idea of overwriting `files.direct`, what do you think of something like this?

```
diff --git a/digdag-ui/model.js b/digdag-ui/model.js
index f0e14d1..0067562 100644
--- a/digdag-ui/model.js
+++ b/digdag-ui/model.js
@@ -252,18 +252,11 @@ export class Model {
     return this.get(`logs/${attemptId}/files?task=${encodeURIComponent(taskName)}`)
   }
 
-  fetchLogFile (file: LogFileHandle) {
+  fetchLogFile (file: LogFileHandle, attemptId: number) {
     if (!file.direct) {
-      return new Promise((resolve, reject) => {
-        reject(`Cannot fetch non-direct log file: ${file.fileName}`)
-      })
+      return this.fetchArrayBuffer(`${this.config.url}logs/${attemptId}/files/${file.fileName}`)
     }
-    return fetch(file.direct).then(response => {
-      if (!response.ok) {
-        throw new Error(response.statusText)
-      }
-      return response.arrayBuffer()
-    })
+    return this.fetchArrayBuffer(file.url)
   }
 
   fetchProjectArchiveLatest (projectId: number): Promise<ProjectArchive> {
@@ -387,6 +380,15 @@ export class Model {
   headers (): Headers {
     return this.config.headers({credentials: this.config.credentials})
   }
+
+  fetchArrayBuffer (url: string) {
+    return fetch(url).then(response => {
+      if (!response.ok) {
+        throw new Error(response.statusText)
+      }
+      return response.arrayBuffer()
+    })
+  }
 }
 
 var instance: ?Model = null
```